src/app.js

```diff
@@ -152,6 +155,24 @@ function exitHandler (options, err) {
   })
 }
 
+// function to update the config
+function updateConfig (configString) {
+  configUtil.setConfig('CHANNEL_FILTER_QUERY_STRING', configString)
+}
+
+// function to generate the search string
+function getFilterConfig () {
+  var allowedChannels = whiteListedChannelList ? whiteListedChannelList.split(',') : []
+  var blackListedChannels = blackListedChannelList ? blackListedChannelList.split(',') : []
+  var configString = {}
+  if (allowedChannels && allowedChannels.length > 0) {
+    configString = allowedChannels
+  } else if (blackListedChannels && blackListedChannels.length > 0) {
+    configString = { 'ne': blackListedChannels }
+  }
+  updateConfig(configString)
```

Comment:
Please move the update out of this function to the call site. Principle of single responsibility.